### PR TITLE
Jrlegrand/30/filters

### PR DESCRIPTION
**Pull request by jrlegrand**
Fixes coderxio/dailymed-api#30, fixes coderxio/dailymed-api#34

## Explanation
Added Django Filters functionality for SPL view and made the SPL view distinct by adding distinct=True to the individual filter elements.  Also made schedule a multiple select and added SCHEDULE_OPTIONS.

## Rationale
So users can filter down to only the SPLs they care about.  Alternatives considered were using GraphQL-type functionality, which we may implement later on.

## Tests
1. Tested out different filters and discovered a duplication bug which was also fixed as part of this PR

### CODE REVIEW

**Comment by finish06** [comment]
@jrlegrand Would you like to fix the styling errors or would you like me to do it?
```
0.53s$ flake8
./api/dailymed/filters.py:4:1: E302 expected 2 blank lines, found 1
./api/dailymed/filters.py:7:80: E501 line too long (100 > 79 characters)
./api/dailymed/filters.py:8:80: E501 line too long (90 > 79 characters)
./api/dailymed/filters.py:9:80: E501 line too long (91 > 79 characters)
./api/dailymed/filters.py:10:80: E501 line too long (125 > 79 characters)
./api/dailymed/filters.py:11:80: E501 line too long (125 > 79 characters)
./api/dailymed/filters.py:12:80: E501 line too long (88 > 79 characters)
./api/dailymed/views.py:11:1: E302 expected 2 blank lines, found 1
```

**Comment by jrlegrand** [comment]
I'll give it a shot tomorrow morning.

**Comment by finish06** [comment]
@jrlegrand Two pending styling errors:
```
$ flake8
./api/dailymed/filters.py:13:1: E302 expected 2 blank lines, found 1
./api/dailymed/filters.py:27:37: W291 trailing whitespace
```

File: api/dailymed/filters.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+
+SCHEDULE_CHOICES = (
+    ('', ''),
+    ('CII', 'CII'),
+    ('CIII', 'CIII'),
+    ('CIV', 'CIV'),
+    ('CV', 'CV')
+)
+
```
Comment (finish06) [inline api/dailymed/filters.py:12]: Add another line break.

File: api/dailymed/filters.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    labeler = filters.CharFilter(
+        field_name='labeler',
+        lookup_expr='icontains',
+        distinct=True)
+    package_ndc = filters.CharFilter(
+        field_name='products__packages__code',
+        lookup_expr='icontains',
+        distinct=True)
+    product_ndc = filters.CharFilter(
+        field_name='products__code',
```
Comment (finish06) [inline api/dailymed/filters.py:27]: Remove white space after `,`

**Review by finish06 — CHANGES_REQUESTED** [review]

**Review by finish06 — APPROVED** [review]
Ran Flake8 locally on pull.  Everything looks good.

**Pull request merged**